### DISCUSSION TITLE
Make force:source:status / pull compatible with multiple users in same org

### DISCUSSION THREAD

**Discussion by jonathanwiesel** [forcedotcom/cli#2259] [Ideas]
**Is your feature request related to a problem? Please describe.**

With new Sandbox Source Tracking capabilities there's the ability the easily determine differences between local and remote metadata, however there are scenarios where an environment could have many developer "touching" at the same time with the intention of each to manage their own changes. 

Taking into account that under the hood this mechanisms are using the [SourceMember](https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_sourcemember.htm) object, which also exposes the `ChangeBy` field, it would be great if we could optionally filter to a specific user changes, not all changes in the org. 

**What are you trying to do**

Make sandbox source tracking more flexible for teams working in the same environment

**Describe the solution you'd like**

The `force:source:status` and `force:source:pull` commands could have a flag to only filter the changes for a particular username. This would imply that the VSCode extension will also need to support this to for no-CLI adepts.

**Describe alternatives you've considered**

Managing this information directly via tooling API or building a custom sfdx plugin based on the original functionality that manages the filtering, but would be best that it would be part of the core since it would grant flexibility so more people can adopt it, never to impose a particular way of working.

[a comment by an automated account is not shown]

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9619564](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000RXxQYAW/view)
